Fixed missions and events to work with escorts.

onpon4 opened the pull request:
Long-standing issue. Many missions assumed that the player would always be the direct attacker, which isn't necessarily true with
escorts. This adds checks for whether or not an attacker's leader is the player as well as whether the attacker is literally the player.

(Note: this comes from Naikari, so if there are any other instances in Naev that were removed there then this won't cover them. But at least in Naikari, this covers all current cases of this mistake in mission/event code.)

EDIT: Ah, and want to note this as well: this fix is important even if escort hiring is never adopted because it affects fighter bay fighters, too.

🕵️

bobbens commented:
Thanks.

> (Note: this comes from Naikari, so if there are any other instances in Naev that were removed there then this won't cover them. But at least in Naikari, this covers all current cases of this mistake in mission/event code.)

I don't find this sort of attitude to be very constructive. You are basically saying you are not interested in anything other than what affects you and have no real intention of collaboration. In the future I would recommend you take a less narrow and egotistical view of open-source development when opening pull requests on projects.

onpon4 commented:
> You are basically saying you are not interested in anything other than what affects you and have no real intention of collaboration.

I'm sorry you got this impression despite my intention being the opposite. But I don't understand your recommendation. Can you explain what exactly you are asking me to do in the future, without making assumptions about my thought process?

🕵️

bobbens commented:
> > You are basically saying you are not interested in anything other than what affects you and have no real intention of collaboration.
> 
> I'm sorry you got this impression despite my intention being the opposite. But I don't understand your recommendation. Can you explain what exactly you are asking me to do in the future, without making assumptions about my thought process?

When you say something like `Note: this comes from Naikari, so if there are any other instances in Naev that were removed there then this won't cover them.`. You are basically saying that the only stuff you care about Naev is that what you use in Naikari. This is generally fine as it is a fork and generally independent, however, if you do a pull request to Naev that affects all of Naev and only modify the parts that affect Naikari, you are not really collaborating in good faith, but just advancing your personal goals. When you do a pull request on Naev, you should act in good faith and modify all the parts that affect Naev, not only Naikari.

onpon4 commented:
> however, if you do a pull request to Naev that affects all of Naev and only modify the parts that affect Naikari, you are not really collaborating in good faith, but just advancing your personal goals. When you do a pull request on Naev, you should act in good faith and modify all the parts that affect Naev, not only Naikari.

So, you're demanding labor from me? I find that rather rude. I don't demand labor from you when you break everything in my setup because you don't pay attention to what I do. What you are essentially telling me is that you don't value me pushing my downstream fixes upstream if those fixes don't happen to cover every possible case in a project I don't even develop for.

🕵️

bobbens commented:
> > however, if you do a pull request to Naev that affects all of Naev and only modify the parts that affect Naikari, you are not really collaborating in good faith, but just advancing your personal goals. When you do a pull request on Naev, you should act in good faith and modify all the parts that affect Naev, not only Naikari.
> 
> So, you're demanding labor from me? I find that rather rude. I don't demand labor from you when you break everything in my setup because you don't pay attention to what I do. What you are essentially telling me is that you don't value me pushing my downstream fixes upstream if those fixes don't happen to cover every possible case in a project I don't even develop for.

I am saying that if you do a PR to Naev which knowingly introduces bugs to Naev and not Naikari, that is not a PR in good faith.

> I find that rather rude. I don't demand labor from you when you break everything in my setup because you don't pay attention to what I do.

First of all, I am not doing pull requests on Naikari, and if I did, I would make sure it didn't break anything in Naikari.

Second, I am not intentionally breaking anything and have told you that as you choose to do things different from Naev that merge conflicts will get worse. I barely have the time and energy to keep up with Naev development let alone follow Naikari development (that basically just undo most of my changes most of the time and aren't really motivating for me personally). I do not want to increase my development overhead on Naev to not break stuff in Naikari, which will break either way because of all the design choices that go in the opposite direction of Naev. I feel that saying that I should not break stuff on Naikari is saying basically that I should be responsible for your choices.

As an example, Nox Imperii actually didn't fork from Naev. It just used Naev as a starting point and went its own way. It also did away with important concepts in Naev by going from a fixed universe to a procedural one, and kinniken decided that it wasn't the effort to fork and keep pulling upstream and just go its own way. It is a shame that development stopped, I would have been interested to see how it turned out.

onpon4 commented:
> I am saying that if you do a PR to Naev which knowingly introduces bugs to Naev and not Naikari, that is not a PR in good faith.

That's not what this PR is. This PR is fixing a long-standing bug in Naev that was literally introduced years ago. Please don't mischaracterize the situation to such an extent.

> Second, I am not intentionally breaking anything and have told you that as you choose to do things different from Naev that merge conflicts will get worse.

Look, I regularly deal with merge conflicts at this point. It's very, very usual for me to find several merge conflicts in a single day. You're really not helping by "explaining" that merge conflicts will happen if we do different things. Of course that will happen. I understand that. Heck, I specifically mentioned divergence just before stepping down as a project member, and you responded by saying that it's not a Naev issue. Which is true, but don't lecture me about something that you specifically told me you don't care about.

> I barely have the time and energy to keep up with Naev development let alone follow Naikari development (that basically just undo most of my changes most of the time and aren't really motivating for me personally). I do not want to increase my development overhead on Naev to not break stuff in Naikari, which will break either way because of all the design choices that go in the opposite direction of Naev. I feel that saying that I should not break stuff on Naikari is saying basically that I should be responsible for your choices.

Great, so you understand my perspective. I feel the exact same way about Naev.

But when I fix a bug that obviously affects Naev too, my natural inclination as someone with a downstream project is to submit a patch upstream. That's what this PR is, and that's what most of my PRs have been in the last week or so. It literally only takes me seconds to do so, but it can save you the effort of having to either look through my commits (which I know you won't do) or, more likely, debug it a second time yourself.

So when I submit such a PR and you respond with hostility because I admitted that I didn't directly test in Naev and it might not literally solve the entire issue that I solved in my project (not that it would hurt Naev, mind, because there's no way this possibly could do that), I feel unappreciated and insulted.

I also feel doubly insulted when you suggest I'm not keeping in mind the repercussions of my contributions despite clear evidence to the contrary. #1799 simply cannot work without two other changes, so I specifically made a point of noting that requirement, even editing the original issue to note the newly introduced requirement when I made changes. And every single fix and improvement I've made to the escort system, I've pushed to the escorts branch, despite the fact that you don't seem interested in the PR, because I don't want you to merge a broken system that I've already fixed downstream. To see that and say that I don't care, to me, feels cold.

🕵️

bobbens commented:
> Look, I regularly deal with merge conflicts at this point. It's very, very usual for me to find several merge conflicts in a single day. You're really not helping by "explaining" that merge conflicts will happen if we do different things. Of course that will happen. I understand that. Heck, I specifically mentioned divergence just before stepping down as a project member, and you responded by saying that it's not a Naev issue. Which is true, but don't lecture me about something that you specifically told me you don't care about.

I would have not brought this up if you had not. In general, I try to avoid beating a dead horse.

> > I barely have the time and energy to keep up with Naev development let alone follow Naikari development (that basically just undo most of my changes most of the time and aren't really motivating for me personally). I do not want to increase my development overhead on Naev to not break stuff in Naikari, which will break either way because of all the design choices that go in the opposite direction of Naev. I feel that saying that I should not break stuff on Naikari is saying basically that I should be responsible for your choices.
> 
> Great, so you understand my perspective. I feel the exact same way about Naev.

Yes, but as I mention, this is your choice not mine. You have chosen to go down a path with Naikari and at the same time try to use Naev as upstream and do all these merges. I am not responsible for you doing this and therefore not responsible for the breakage I may cause because I don't follow Naikari. If you are tired about merge conflicts and such, the solution is simple, just don't pull. Getting angry at Naev for breaking stuff when you are the one diverging and you are the one pulling is something I just can't see how it is a Naev issue. Given the amount of small changes all over, I honestly don't think there is any way to make this not be an issue and I do think full divergence is pretty much inevitable.

I do make an effort for compatibility when possible such as total conversions, however, they are a completely separate issue from Naikari as they use the Naev engine as-is and only aim at replacing all the data files which is something I do think supporting is good. However, as mentioned, I just don't see how it is possible to make Naev and Naikari development work together given that instead of replacing all the data, it is doing a lot of more smaller changes which inevitably make merging impossible.

> But when I fix a bug that obviously affects Naev too, my natural inclination as someone with a downstream project is to submit a patch upstream. That's what this PR is, and that's what most of my PRs have been in the last week or so. It literally only takes me seconds to do so, but it can save you the effort of having to either look through my commits (which I know you won't do) or, more likely, debug it a second time yourself.
> 
> So when I submit such a PR and you respond with hostility because I admitted that I didn't directly test in Naev and it might not literally solve the entire issue that I solved in my project (not that it would hurt Naev, mind, because there's no way this possibly could do that), I feel unappreciated and insulted.

I have never said I don't appreciate your PRs and have always thanked you and encouraged you to submit. However, I feel that when you do a PR that knowingly introduces bugs in Naev and not Naikari, it takes time and effort to check all the cases and make sure it works in Naev. This is not really what the PR framework is for. I guess if you mark it as a `draft` it could work, however, that is misleading because it sort of indicates that you still intend to work on it. Maybe either some sort of PR tag to indicate that this is untested and mention possible issues, or just raising an issue with a link to a possible solution would be the best solution in this case.
 
> I also feel doubly insulted when you suggest I'm not keeping in mind the repercussions of my contributions despite clear evidence to the contrary. #1799 simply cannot work without two other changes, so I specifically made a point of noting that requirement, even editing the original issue to note the newly introduced requirement when I made changes. And every single fix and improvement I've made to the escort system, I've pushed to the escorts branch, despite the fact that you don't seem interested in the PR, because I don't want you to merge a broken system that I've already fixed downstream. To see that and say that I don't care, to me, feels cold.

I never said you are not keeping in mind the repercussions. You generally do a thorough job in your PRs and I am grateful. To reiterate, I simply said that when you submit a PR that knowingly introduces bugs into Naev but not Naikari, that does not seem to be in good faith.

> despite the fact that you don't seem interested in the PR

I don't know where you get this idea. I have repeatedly mentioned that escorts is something that we want, I just think that we have to consider many different implications, which I have raised again and again in different issues and such. Currently, I am still working on the rebalance stuff and haven't even begun to look and consider the escort PR yet.

Maybe I have overreacted, and for that I am sorry. It is probably me interpreting your lack of interest as complicit malice. I have never intended to make you feel like your contributions are worthless. However, I do still find your attitude frustrating as a Naev maintainer, but I guess I will try to work on understanding your issues better. For now, as a solution, I have added a `PR-Untested` label. Please use it for PR like this one that may introduce bugs in Naev and not Naikari, so it is clear that it shouldn't be pulled as is without in-depth checking.

onpon4 commented:
> If you are tired about merge conflicts and such, the solution is simple, just don't pull.

Perhaps I should have been more specific. When I said I feel the same way, it's about the amount of responsibility I have for the Naev project. I'm not frustrated by the task of merging or, as you say, I would stop. I probably will stop eventually but haven't made it to the point where I want to yet.

> I have never said I don't appreciate your PRs and have always thanked you and encouraged you to submit. However, I feel that when you do a PR that knowingly introduces bugs in Naev and not Naikari, it takes time and effort to check all the cases and make sure it works in Naev. This is not really what the PR framework is for.

Except this PR, where you decided to bring this up, isn't introducing bugs in Naev, it's fixing bugs in Naev that have been in Naev for years. I have never knowingly introduced bugs into the Naev project, and in fact every time I ever played Naev I've always come forward with several bugfixes that I've come across. That's exactly what this PR is, coming across a bug and fixing it. In this case I came across it because I was implementing the escort system, but remembered that it also affected fighter bay fighters, so I opened a PR with my fixes (this PR). The fixes use an API that originated in Naev and is currently in Naev, so there simply is no possible way that this could cause a breakage unless I made a syntax error or something, in which case it would break downstream too.

> However, I do still find your attitude frustrating as a Naev maintainer, but I guess I will try to work on understanding your issues better.

I honestly have no idea what "attitude" you are attributing to me here.

I'll be the first to admit that I have a lot of problems, but so far you're talking entirely about things that I simply don't do. The closest I have ever come to "knowingly introducing bugs" is the rebalance branch. where simply because of the type of work it is it probably won't work the same way in Naikari and Naev. But I've been very open about the fact the fact that Naikari is my testing bed for that branch, and the points where it can possibly be weak in the balance department have to do with stuff you have recently unilaterally added to Naev that I am unfamiliar with, like those special super-weapons or whatever and Lua outfits. So even this isn't "knowingly introducing bugs", it's Naev marching along ahead of me because it took me a long time (an unreasonably long time, admittedly) to finish it.

And this PR, #1801, certainly isn't "knowingly introducing bugs". It's *fixing* bugs knowing that it might not fix the bug literally everywhere. I fail to see the difference between this and, say, only fixing typos in the FLF campaign and noting, "by the way, this is only fixing the FLF campaign, so there might be other typos elsewhere". Was it unnecessary to say? Probably, but it certainly doesn't have anything to do with "knowingly introducing bugs".

> For now, as a solution, I have added a PR-Untested label. Please use it for PR like this one that may introduce bugs in Naev and not Naikari, so it is clear that it shouldn't be pulled as is without in-depth checking.

That's not possible. Only project members can add tags.

🕵️

bobbens commented:
> > However, I do still find your attitude frustrating as a Naev maintainer, but I guess I will try to work on understanding your issues better.
> 
> I honestly have no idea what "attitude" you are attributing to me here.

This could be me, but in general, I find you have trouble considering people's viewpoints or even acknowledging them. I can give you several examples:

1. Time compression you set was very high. Several people on discord, including developers, thought this was an issue and we discussed lowering. You seemed to not acknowledge that this could be an issue despite several people pointing it out it was, and as far as I know have reverted the change of halving it that we did in your fork.
2. When you changed the viewing slot sizes bg in the image array of the outfits for colorblind accessibility, several people pointed out that they preferred it when it had colour. You never really seemed to acknowledge that this could be a problem and recently we changed the colours so that it is still colorblind accessible and makes more users happy.
3. When discussing the rebalance branch, I mentioned that it was weird that there were so many core outfits that were virtually the same, and in some cases, there were objectively worse ones. You then used a strawman argument to say that why not just do one core per ship slot and didn't even consider that this could be an issue. Afterwards you apologized and removed the B-X platings.
4. I developed the VN system over some months due to complains I aired with the `tk.msg()` system for missions citing how dry it is. You basically said the VN system is the "worst thing ever done to Naev" while never really saying anything that could be done to improve it (other than just remove it), nor actually acknowledging the fact that some people may find the `tk.msg()` system not engaging enough.
5. When adding the travelling merchant and unique outfits. You went on to argue against me that unique outfits have no place in Naev nor that it wants to be an RPG despite unique outfits being listed in the TODO file, and RPG being on the description of Naev everywhere. Later on you went to say that "I have no development direction and have no idea what I'm doing" or something like that.

In general, even though I try to keep an open approach to Naev development and discuss things as much as possible, I do get very frustrated with our conversations and don't find them to be very productive. I always get the feeling that you just don't want to compromise or even acknowledge issues I raise, and don't really provide constructive criticism in many cases. The conclusion is always the same: "I'll do whatever I want in Naikari".

As this is open-source, you are completely free to do what you want and I encourage you to do so, but you constantly talk about Naikari and make all these comparisons with Naikari all the time when it is not really related to the topic in hand. If you do a PR and say "I only tested this on Naikari and don't know what it is like in Naev", that is not an issue (in this aspect), but if a new person comes on discord and asks a #gameplay question and you say something like "Naev does XXX, but we do YYY in Naikari", that can just be confusing to new players who came to ask about Naev and don't know what Naikari is.

I guess one of the issues is I don't really know what role you want Naev to play in Naikari, and what you want Naikari to be.

> And this PR, #1801, certainly isn't "knowingly introducing bugs". It's _fixing_ bugs knowing that it might not fix the bug literally everywhere. I fail to see the difference between this and, say, only fixing typos in the FLF campaign and noting, "by the way, this is only fixing the FLF campaign, so there might be other typos elsewhere". Was it unnecessary to say? Probably, but it certainly doesn't have anything to do with "knowingly introducing bugs".

Yes, as I mentioned, I overreacted by extrapolating to not-yet existent cases (such as changing API and only changing stuff that affects Naikari) and am sorry. This is probably caused by pent-up frustration with our interactions and I will work to address it.

> > For now, as a solution, I have added a PR-Untested label. Please use it for PR like this one that may introduce bugs in Naev and not Naikari, so it is clear that it shouldn't be pulled as is without in-depth checking.
> 
> That's not possible. Only project members can add tags.

That is a shame.

onpon4 commented:
> You seemed to not acknowledge that this could be an issue despite several people pointing it out it was, and as far as I know have reverted the change of halving it that we did in your fork.

Half of what you say here is true. I was strongly opposed to lowering those values, perhaps unreasonably so.

> When you changed the viewing slot sizes bg in the image array of the outfits for colorblind accessibility

No, that wasn't me. That was actually a different contributor I merged the pull request of, incidentally a contributor who was frustrated that you were rolling back many of their changes.

> You never really seemed to acknowledge that this could be a problem and recently we changed the colours so that it is still colorblind accessible and makes more users happy.

It's not accessible to full colorblindness; two of the colors look nearly identical under rod monochromacy. This is a minor issue now because slot size is shown in a popup (though having an obvious visual indicator in the form of color is such a huge boon that being distinguishable is really helpful). But the reason I reverted that change downstream is because I like the all-blue color scheme aesthetically and am not a fan of the candy colors.

> I developed the VN system over some months due to complains I aired with the tk.msg() system for missions citing how dry it is. You basically said the VN system is the "worst thing ever done to Naev" while never really saying anything that could be done to improve it (other than just remove it), nor actually acknowledging the fact that some people may find the tk.msg() system not engaging enough.

The VN system was one of many things you unilaterally introduced to Naev with no community input in the last year or so. That was really frustrating to deal with when I was still a contributor to Naev and is entirely why I forked into my own project.

That said, my jabs at you were uncalled for, and I'm sorry. In fact this has been a problem with me since long before that, and I understand that I caused unnecessary hurt by doing so. I'm sorry for that, and I will make an effort to stop personally attacking people I'm frustrated with in the future.

> When adding the travelling merchant and unique outfits. You went on to argue against me that unique outfits have no place in Naev nor that it wants to be an RPG despite unique outfits being listed in the TODO file, and RPG being on the description of Naev everywhere. Later on you went to say that "I have no development direction and have no idea what I'm doing" or something like that.

Similar to the above, I'm sorry for this.

There were many, many other instances where I rudely personally attacked you for disagreements, some of them minor, and that was unacceptable. I can't think of any way I could have civilly discussed these decisions you made on your own, but it's your project, so I should have just accepted what you're doing and left the Naev development team immediately when I knew things were going in a direction I wasn't okay with. I considered doing so, but I didn't, and I instead said unreasonably hostile and passive-aggressive things to you. I'm sorry that I did that.

> In general, even though I try to keep an open approach to Naev development and discuss things as much as possible, I do get very frustrated with our conversations and don't find them to be very productive. I always get the feeling that you just don't want to compromise or even acknowledge issues I raise, and don't really provide constructive criticism in many cases. The conclusion is always the same: "I'll do whatever I want in Naikari".

While some of my behavior was unacceptable, I cannot agree with the statement that you "discuss things as much as possible". Changes you started making unilaterally a year ago blindsided me. You mention the unique outfits being listed in the todo file or something, which I don't particularly remember, but what I do remember is that any time you had an idea you would immediately make it the plan for the future, and meanwhile if I had an idea, you would say that actually, the plan is X (where X is something not noted anywhere). You've also accepted contributions with no plan of integrating them into the larger game, like when you marged my contribution granting access to Thurion space through the FLF and then months later said something about that just being a "temporary solution", something which you had never communicated prior to that.

Prior to about a year ago, my experience with you in this project is that you've been a passive gatekeeper, knowing almost nothing about your own game while blocking progression on ideas you didn't approve of. You still do this, and that's valid. My problem started when you started making your own changes without applying that same standard to yourself, so many changes I've barely been able to keep up with them let alone discuss whether or not they're good. Of course there are times where you do apply the standard of appeasing the whole team before implementing something, but that didn't make the many times you introduced something only because you thought it was a good idea (without consulting anyone else so far as I can tell) weren't made less irritating by that fact.

> but if a new person comes on discord and asks a #gameplay question and you say something like "Naev does XXX, but we do YYY in Naikari", that can just be confusing to new players who came to ask about Naev and don't know what Naikari is.

I think I remember doing that once or twice and I apologize for that, though I think that was a long time ago. I won't be making confusing statements like that in the future.

> I guess one of the issues is I don't really know what role you want Naev to play in Naikari, and what you want Naikari to be.

Weirdly enough that is one of the problems I have with you: I have no clue what you want to do with Naev and when you started making decisions all on your own about stuff, it made me feel like I couldn't trust you anymore.

The direction I'm going with Naikari is the same direction I always drove Naev towards: a freeform sandbox space trading and mercenary game. RiftSpace and Star Wraith 2's mercenary mode are my primary influences.

🕵️

bobbens commented:
> Prior to about a year ago, my experience with you in this project is that you've been a passive gatekeeper, knowing almost nothing about your own game while blocking progression on ideas you didn't approve of. You still do this, and that's valid. My problem started when you started making your own changes without applying that same standard to yourself, so many changes I've barely been able to keep up with them let alone discuss whether or not they're good. Of course there are times where you do apply the standard of appeasing the whole team before implementing something, but that didn't make the many times you introduced something only because you thought it was a good idea (without consulting anyone else so far as I can tell) weren't made less irritating by that fact.

In general, I have been accepting almost all changes to Naev, and I do believe I try hard to incorporate as many peoples ideas as possible. Could you give me some examples?

I think the main issue I have, is that I've tried to have Naev be a positive community-driven project as much as possible. The idea being that everyone is able to give and share ideas and get constructive criticism on it to see how it can be made work with Naev. In a few cases things are not possible and stuff like Nox Imperii gets forked, but in general, I do pull and merge stuff that I am not entirely convinced of because I vouch for the person proposing it and think that diversity is what makes Naev great. Thanks to this we have gotten lots of things I had never thought of or expected and I think that the positive discussion environment is what makes Naev great. However, I do feel like lately this has been taken away from me and the Naev community.

When I developed the travelling merchant and the accessories, for which I did ask for feedback and pushed to a separate branch instead of master directly, I thought that this would be something that players could and enjoy and would be interesting to have. At this time you were really strongly against it, but instead of offering any constructive criticism like what about changing it to XX or doing YY, you just said it was horrible and not Naev. Afterwards this disagreement led to you forking and creating Naikari, despite the accessories and travelling merchant being something completely optional you could ignore without any significant effect on anything. For me this basically is a strong signal that you do not wish to compromise, and personally feel that this goes everything that I try to do with my community-based approach in Naev and for this reason I dislike Naikari, however, I don't have the same issue with any of the TC nor Nox Imperii because they are fundamentally different games. I don't want you to misunderstand the fact that I do think you have forked Naikari because you feel very strongly and are passionate, but I still have serious issues taking it as just not willing to compromise.

The thing is, in open source development, we are always going to have disagreements. There are many aspects of Naev I don't really like or agree with, one of the most notable being the Thurion, which I think just don't really fit into the Naev universe as is (but I do think it is fixable, and is a topic out of this scope), however, I do think that it is important to try to work and push together, and try to do this in a constructive manner. If people feel that one of your implemented mechanisms is bad or needs improving, you sit down and try to fix it like how missiles regenerate or you don't have to transport your ships around anymore. If someone wants space unicorns, you say "maybe it's complicated, but how about some Soromid NPCs that have horse DNA and a single horn?", and try to make things work together.

The issue I have lately, is I just don't feel motivated to discuss things as much as I used to. I am trying to, but as an example in the rebalance PR, I tried it for several hours, wrote down a list of issues and thoughts I had and then gave them to you. You basically dismissed all of them and I felt I had to be super defensive about them, even though at the end you did apologize and slightly change something related to one issue (removing B-X plating). However, this came after several hours of frustrating discussion where I felt basically none of my issues had even been addressed. This is just one example, but this is something that has been happening continuously. Later, I even pushed some minor changes (soromid missing bioweapon and za'lek hunter/reaper drones too strong), and you literally reimplemented the za'lek stuff instead of even looking at what I did first. Since then I have played even more and done lots of numbers and computations on how to improve issues and aspects I find with the rebalance branch, and been pushing them, but honestly I just abhor the idea of having to discuss them with you and am basically unilaterally implementing them. I guess I will just ask for feedback from the Naev community down the line.

To summarize a bit, I guess the entire reason I am frustrated lately is a combination of several factors. First, when I present an idea or whatever, I feel that I have to get very defensive when it is something you feel strongly about, and since you generally don't present any options or constructive criticism that deals with the underlying issue, I generally just go ahead with it. This causes you frustration because you think I am unilaterally doing things (which in part I am), and makes you be more aggressive in your arguments which in turn makes me more defensive and less inclined to actually discuss stuff. As this cycle repeats, everyone is unhappy with the outcome and this is not good for anyone.

So the question is, how to improve the situation? I am fine with you doing and working with Naikari, even though I don't like Naikari as mentioned above, and chances are I never will. For my part, I will try to be more comprehensive to your situation (like the lack of delicacy in some of my replies in this issue) and discuss more of the things I intend to do. However, on your side, I would like you focus on constructive criticism and if possible trying to take into account issues other people have instead of being dismissive of most feedback. If you have any other comment or thing you would like to see on my part, feel free to mention it.

onpon4 commented:
> In general, I have been accepting almost all changes to Naev, and I do believe I try hard to incorporate as many peoples ideas as possible. Could you give me some examples?

By "gatekeeper", I mean that you checked my work before merging it and at times asking me to change things you didn't like. I can find specific examples if you don't think you've ever done that, but I get the impression you interpret "gatekeeper" to mean you're unwilling to accept changes at all, which isn't how I was using the term. Sorry for my confusing word choice.

> When I developed the travelling merchant and the accessories, for which I did ask for feedback and pushed to a separate branch instead of master directly, I thought that this would be something that players could and enjoy and would be interesting to have. At this time you were really strongly against it, but instead of offering any constructive criticism like what about changing it to XX or doing YY, you just said it was horrible and not Naev.

I don't buy that adding things is always good and it's wrong to outright reject a concept because you think it would detract from the work as a whole. I'll give an example: if someone opened an issue saying that Naev needs swastikas in its title screen, you wouldn't discuss how to better use swastikas, you would close the issue and tell them in no uncertain terms that swastikas do not belong in Naev.

That's an extreme example chosen because I'm sure you would agree with me on this, not to compare any of your changes to that hypothetical. But this is my perspective: in a cohesive game, some things just don't belong. I disagree with your approach of always approaching suggestions as things to figure out *how* to add. *If* those suggestions should be added is also important to consider. So I think it's unreasonable to mark any opinion that is completely against an idea as not "constructive".

The reason I'm against traveling merchants as a concept is because they fundamentally interrupt a system where a player can see on the map where they can get certain things, and they have to travel around to get what they want. A traveling merchant being added means, depending on how it's implemented, either: the merchant is unpredictable and you get access to it randomly; the merchant is always available and you always get access to outfits; or the merchant is consistently in certain system and it acts like an additional planet the map doesn't keep track of.

I fail to see how I'm being unconstructive by having this opinion. On your side, I still have no clue what problem the traveling merchant was intended to solve despite it being very important to you. There was no discussion I was a part of where some problem was identified and traveling merchants were created to solve that problem. It was just something you threw out there, and then you outright refused to consider rejection of the idea as an option.

> despite the accessories and travelling merchant being something completely optional you could ignore without any significant effect on anything.

I'm not approaching this from the perspective of a player, I'm approaching it from the perspective of a developer. From that perspective, optional things have real consequences, namely affecting what kind of game it is.

> There are many aspects of Naev I don't really like or agree with, one of the most notable being the Thurion, which I think just don't really fit into the Naev universe as is

Funny enough, the Thurion are one of my favorite parts of the game, and I have no idea why you would dislike them. I don't think it's particularly fair or productive to just go along with something you specifically don't like if you're just going to change it somewhere down the line at a whim, so that you can "make things work together". That's not working together in my opinion, that's just ignoring disagreements so they can become a problem later.

> If people feel that one of your implemented mechanisms is bad or needs improving, you sit down and try to fix it like how missiles regenerate or you don't have to transport your ships around anymore.

Incidentally, I'm responsible for both of those. Other things I'm responsible for are making autonav not cancel the second an enemy shows up, making the Brushed GUI show more information, adding text outlines, adding symbols for colorblind accessibility, fixing the colorblind mode pretty much from the ground up to make it more accurate and capable of emulating more colorblindness types, adjusting colors to make them better and more colorblind-accessible, and going through almost every single mission over the years and vetting them for language mistakes, just off the top of my head.

I assume you brought those two things up knowing that I'm responsible for them to point to examples of this type of thing that I've done right, but your tone implies that I'm not aware of this, and my contribution history clearly demonstrates that I'm very much aware and I've done things like that many, many times. Heck, I'm pretty sure the vast majority of my commits have been fixes.

> but as an example in the rebalance PR, I tried it for several hours, wrote down a list of issues and thoughts I had and then gave them to you. You basically dismissed all of them and I felt I had to be super defensive about them, even though at the end you did apologize and slightly change something related to one issue (removing B-X plating). However, this came after several hours of frustrating discussion where I felt basically none of my issues had even been addressed.

If you're going to bring this up again, let me explain that that entire time, I had no clue what problems you were trying to solve with your suggestions. All I had to go off of was your word that it would be a good idea. I made those changes when I came up with an actual reason to make them: I realized that the Unicorp D platings and low-level engines were underused, and I also noticed that cutting down on the number of engines would mean less image duplication. So I eliminated the B- platings specifically to encourage use of D- platings instead, and I squished low-level engines down into a mix of what Unicorp and Nexus engines were specifically to make low-level engines more likely to be useful, with the nice side-effect of making most of the engines have unique shop graphics. None of this was anything you brought up in the issue. You just told me your intuition that there were too many outfits. The result was that your suggestions sounded to me more like a list of demands.

This doesn't excuse the fact that I got into an unproductive argument with you, but in that sort of framing, while I could have made it less frustrating for you by just ignoring you, that wouldn't have by itself made the discussion productive. The only way I can think of that I could have done what you wanted me to do, or even considered it, without causing that argument would have been to just do what you say without question, which I don't think is reasonable to expect.

> Later, I even pushed some minor changes (soromid missing bioweapon and za'lek hunter/reaper drones too strong), and you literally reimplemented the za'lek stuff instead of even looking at what I did first.

You made changes in a series of three commits that were difficult to view in totality to fully parse, so that's one reason I didn't look at it. The other reason is because those commits were to solve a bug, which directly stemmed from me not noticing a couple of values that I already knew were not the ones you adjusted, so I opted to fix the bug the obvious way (based on how I intended those outfits to work) rather than evaluate whether or not your change had the same overall effect.

> However, on your side, I would like you focus on constructive criticism and if possible trying to take into account issues other people have instead of being dismissive of most feedback.

Given what you've established "constructive" to be, what this ultimately translates to, so far as I understand, is never expressing opposition to ideas. I won't be particularly happy with that, but I'll comply with it if that's what you really want.

🕵️

bobbens commented:
> ...

I only see you are basically making my points with added Godwin's law as a bonus.

> Given what you've established "constructive" to be, what this ultimately translates to, so far as I understand, is never expressing opposition to ideas. I won't be particularly happy with that, but I'll comply with it if that's what you really want.

You know you are doing a straw man argument again, but at this point I would gladly take your suggestion over what we have now, so, for the sake of my mental health, yes please.
